### ISSUE TITLE
`--createstub` doesn't output the `/` "positional only" separator

### ISSUE THREAD

**Issue by mniip** [microsoft/pyright#3406]
**Describe the bug**
All function signatures containing a `/` separator aren't printed correctly by `--createstub`, yielding syntactically invalid python.

**To Reproduce**
Run `pyright --createstub` a on the `a.py` file from the Code section. Observe that the resulting `pyi` file is incorrect.

**Expected behavior**
The `/` should be printed correctly.

**Screenshots or Code**
```py
# a.py
def foo(x, /, y): ...
bar: lambda x, /, y: x
```
```py
"""
This type stub file was generated by pyright.
"""

def foo(x, , y): # -> None:
    ...

bar: lambda x, , y: x
```

**VS Code extension or command-line**
Commandline pyright 1.1.241

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by mniip** [comment]
I am not sure how often `lambda` can appear in types, but the pretty-printer for lambdas is still broken (reproducible by the file I provided in the issue), c.f.:

https://github.com/microsoft/pyright/blob/3b5752431b5e5d52a31cabe9920c8d60395566f3/packages/pyright-internal/src/analyzer/parseTreeUtils.ts#L378-L399

**Comment by erictraut** [comment]
Lambdas never appear in type annotations and shouldn't be used in stubs. Do you have a case where you're seeing this?

The code sample you provided at the top of this bug report would be rejected because a lambda expression is an invalid type expression.

I suppose it's possible to use a lambda as a default argument, so I'll fix this in the type printer.

**Comment by mniip** [comment]
No, I don't have an actual use case. The code sample was just the first thing I could come up with to trigger the code path for printing lambdas.

The code sample was not rejected by `--createstub` so I included it in the MWE.

Either case I am sufficiently happy now.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.242, which I just published. It will also be included in the next release of pylance.
